fea-rs/src/parse/context.rs

```diff
@@ -127,14 +116,11 @@ impl ParseContext {
     pub fn parse_from_root(
         path: PathBuf,
         glyph_map: Option<&GlyphMap>,
-        project_root: Option<PathBuf>,
-    ) -> Result<Self, HardError> {
-        let source = Source::from_path(path).map_err(|e| HardError::IoError {
-            path: e.path,
-            cause: e.cause,
-        })?;
-        let mut sources = SourceList::new(source, project_root);
-        let mut queue = vec![sources.root_id()];
+        resolver: impl SourceResolver + 'static,
```

Comment:
noob question, why 'static? - that feels like an odd constraint

Reply:
This is a common bound you will see on types that are going to end up as `Box<dyn trait>`; historically only types with a `'static` bound could live on the heap. I believe this has now changed such that you can have a type on the heap with a single named lifetime, but there is no meaningful lifetime associated with this trait, so we stick with `'static`.

Reply:
Ty, I had not encountered https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html#trait-bound yet